src/Microsoft.Extensions.FileProviders.Physical/PhysicalFileProvider.cs

```diff
@@ -57,14 +57,49 @@ public void Dispose()
 
         private string GetFullPath(string path)
         {
+            if (PathNavigatesAboveRoot(path))
```

Comment:
Wouldn't `IsUnderneathRoot` take care of rooting this? If not, could you add a comment about scenarios this method prevents that `IsUnderneathRoot` does not?